content/blog/taking-control-of-plot-scaling/index.Rmd

```diff
@@ -0,0 +1,182 @@
+
+The first thing we need to do is convert our physical dimensions to pixel dimensions using our desired resolution. We want to end up with a 60x36cm plot at 300ppi. This gives us:
+
+```{r, message=FALSE, out.width=""}
+c(60, 36) *
+  0.3937 * # convert to inch
+  300 # convert to pixels
+```
+
+We can now use these values directly in our device and change the resolution of the device to trick it into thinking that text etc should be rendered at a larger size
+
+```{r, message=FALSE, out.width=""}
+pngfile <- fs::path(knitr::fig_path(),  "resolution.png")
+agg_png(pngfile, width = 7087, height = 4252, units = "px", res = 900)
+plot(p)
+invisible(dev.off())
+knitr::include_graphics(pngfile)
+```
+
+This actually works exactly as we hoped. We've gotten our huge version of the plot but with the same exact scaling of all graphic elements.
+
+Depending on your temperament this may be a perfect solution. To me, I think the conversion from physical dimensions to pixels is tedious, and it has the added drawback that the dimensions of the plot is incorrectly encoded (it will appear as a 20*12cm plot at 900ppi) which may impact how it is presented in different programs. In the end you may have to manually resize it to get the correct physical dimensions in the end.
+
+## The solution
+Seeing that there is no single perfect solution to fixing this with the tools at our disposal, I've added a new argument to the ragg devices called `scaling`. It's a multiplier that is applied to all absolute sizes, without interfering with the encoded dimensions of the output. Since we have increased the dimensions 3 times we set `scaling = 3` to make sure that the absolute sized elements are keeping their relative size.
+
+```{r, message=FALSE, out.width=""}
+pngfile <- fs::path(knitr::fig_path(),  "scaling.png")
+agg_png(pngfile, width = 60, height = 36, units = "cm", res = 300, scaling = 3)
+plot(p)
+invisible(dev.off())
+knitr::include_graphics(pngfile)
+```
+
+As can be seen, the new argument makes it very easy to reclaim the look of the plot after resizing. Hopefully this will remove a good deal of the pain related to generating plots for papers, posters, presentations, etc.
+
+## Addendum
+- The discussion above is especially relevant to raster output since they don't resize gracefully and it is thus very important to get the correct dimensions and resolution when it is rendered. However, rendering to a pdf or svg yields some of the same issues. For the `pdf()` device the only way to correct it is to render it at the dimension where it looks good, and then manually resize it once you've imported it into wherever it needs to be. Since it is vector based the resizing will not result in quality issues. As for svgs, the next version of the svglite package will include the same scaling argument as the ragg devices, which means that the given solution is applicable there.
```

Comment:
I think it's worth mentioning this earlier — the easiest fix for the motivating problem is just to save as a pdf and make 3x bigger for the poster. Then the rest of the post can focus on raster devices.